src/sss_client/common.c

```diff
@@ -1039,19 +1040,19 @@ static errno_t check_server_cred(int sockfd)
         return errno;
```

Comment:
You need to read proper socket option here depending on HAVE_UCRED/HAVE_XCRED: FreeBSD man page says it is 'LOCAL_PEERCRED' there.

Reply:
```
$ grep -rn SO_PEERCRED *
responder/common/responder_common.c:115:    ret = getsockopt(cctx->cfd, SOL_SOCKET, SO_PEERCRED, &cctx->creds->ucred,
sss_client/common.c:1036:    ret = getsockopt(sockfd, SOL_SOCKET, SO_PEERCRED, &server_cred,
tests/intg/getsockopt_wrapper.c:122:    if (ret == 0 && level == SOL_SOCKET && optname == SO_PEERCRED
```

Maybe you don't worry about intg-tests though.

Reply:
I'm not sure if `LOCAL_PEERCRED` and `struct xucred` correlate, so instead just used `#ifdef LOCAL_PEERCRED` instead.

Reply:
I think it's pretty reasonable to expect they do, because option and structure that this option populate are tightly related.

Reply:
We could also add something like SSS_PEERCRED_SOCKET_OPTION in 'util_creds.h' to avoid #ifdef in every place it's used.